.github/workflows/cleanup-preview.yml

```diff
@@ -0,0 +1,24 @@
+name: Cleanup Preview
+
+on:
+  pull_request:
+    types: [closed]
+
+permissions:
+  deployments: write
+  pull-requests: write
+
+jobs:
+  cleanup:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      
+    - name: Delete deployment environment and comment
+      uses: actions/github-script@v7
+      with:
+        script: |
+          const cleanupScript = require('./.github/scripts/cleanup-preview.js');
+          await cleanupScript({ github, context });
```

Comment:
That is suspicious, I wouldn't think you can await outside of anything like that.